README.md

```diff
@@ -4,18 +4,236 @@
 * 본 레포지토리는 백엔드 스터디 2-3주차의 과제를 위한 레포입니다.
 * 따라서 해당 레포를 fork 및 clone 후 local에서 본인의 깃헙 ID 브랜치로 작업한 후 커밋/푸시하고,
 PR 보낼 때도 `본인의 브랜치-> 본인의 브랜치`로 해야 합니다.
- 
+
 ## 2주차 과제 (기한: 9/26 토요일까지)
 노션 링크 (https://www.notion.so/2-Django-ORM-c46e2d2f88ac4d948d012c07605d8e03)
 
 ### 서비스 설명
-본인이 선택한 서비스에 대한 설명을 적어주세요!
+
+영화 목록 서비스로, 현재 상영 중인 영화와 이에 대한 정보(리뷰, 평점, 등장배우, 설명서.. 등)을 볼 수 있도록 하는 서비스이다. 현재는 모델링만 했지만, 좀 더 구현해서 User에 대한 부분도 추가하도록 노력해볼 것이다.
+
+기능의 경우 역시 아직 구현하지 않았지만, 기존에 존재하는 영화 어플처럼 사용자가 영화를 구매하고 평점과 리뷰를 달아 그것들을 볼 수 있게 해주는 서비스를 구현해보려한다.
+
+
 
 ### 모델 설명
-서비스에 대해 본인이 작성한 모델들에 대한 설명과 모델 간의 관계 등을 적어주세요!
+
+Actor
+
+```python
+class Actor(models.Model):
+    name = models.CharField(primary_key='TRUE', max_length=20)
+    age = models.IntegerField()
+    height = models.FloatField()
+```
+
+- name을 기본 키로 설정하고 그 외 배우에 대한 신상 정보를 넣었다.
+
+Movie
+
+```python
+class Movie(models.Model):
+    # Actor와 Movie를 M:N으로 연결
+    actor = models.ManyToManyField(Actor)
+    title = models.CharField(max_length=30)
+    house = models.PositiveIntegerField(unique='TRUE', default=0)
+    price = models.IntegerField(default=10000)
+    seat = models.IntegerField(default=100, null='TRUE')
+
+    def __str__(self):
+        return self.title
+```
+
+- 해당 서비스에서 가장 중심이 되는 모델
+- Actor와 M:N 구현을 했다. 따라서 각 영화 당 등장 배우들을 볼 수 있고, 반대로 각 배우 당 참여한 영화들을 볼 수 있다.
+
+Comment
+
+```python
+class Comment(models.Model):
+    movie = models.ForeignKey(Movie,
+                              on_delete=models.CASCADE,
+                              related_name='comment')
+    review = models.TextField(max_length=1000)
+    written_date = models.DateField(default=timezone.now)
+    """ 평점에 대한 부분은 0~10으로 제한했고, validators.py를 통해 구현 """
+    grade = models.IntegerField(default=5, validators=[validate_score])
+    count = models.IntegerField(default=0)
+
+    def __str__(self):
+        return '<%s, %s>' % (self.movie.title, self.review)
+```
+
+- Movie의 외래 키로 설정했다. 영화 한 편에 많은 코멘트가 생길 수 있으므로 영화와 1:M 관계를 구축했다.
+
+- 개선해야 할 점
+
+  - grade에 대한 부분: default를 validators=[validate_score]
+
+    ```python
+    # validators.py
+    from django.core.exceptions import ValidationError
+    
+    
+    def validate_score(value):
+        """평점(score)이 10보다 크면 Validation Error 를 일으킨다."""
+        if (value > 10) | (value < 0):
+            msg = u"'평점은 0 이상 10 이하로 매겨주세요."
+            raise ValidationError(msg)
+    ```
+
+    으로 구현했는데, 원했던 점은 0~10 외의 grade가 입력되면 오류가 나도록 하는 것이었다.
+
+    ```python
+    >>> c3=m1.comment.create(review="review3 of movie1", grade=11, count=3) # 오류가 나길 간절히 빕니다.
```

Comment:
https://docs.djangoproject.com/en/3.0/ref/validators/#how-validators-are-run
를 참고해보시면 아래 내용을 확인하실 수 있습니다.

```Note that validators will not be run automatically when you save a model, but if you are using a ModelForm, it will run your validators on any fields that are included in your form```

model instance의 validating logic을 호출하기 위해선 아래 도큐먼트를 참고해보시면 도움될 것 같아요
https://docs.djangoproject.com/en/3.1/ref/models/instances/#validating-objects